### PR TITLE
Correct Enterprise upload url

### PR DESCRIPTION
**Pull request by SteelPhase**
The upload endpoint is not `/api/v3` it is `/api/upload`

Tested against Github Enterprise

### CODE REVIEW

**Comment by gmlewis** [comment]
Can you please find an Enterprise documentation link and add that to a comment where these changes are located?

**Comment by SteelPhase** [comment]
I've been looking for documentation, and I'll update this comment when I find it.
I'm not sure the url was ever correct, as I've been using a couple years with `/api/uploads` on Github Enterprise.

**Comment by SteelPhase** [comment]
I'm going to throw in a commit to update the description

**Comment by gmlewis** [comment]
Oh, and no need to force-push in this repo. We always "Squash and merge".  Thanks.

**Comment by SteelPhase** [comment]
Oh good to know. I was wary and checked another PR, but thanks for letting me be slightly commit messy!

File: github/github.go [hunk 1]
```diff
@@ -291,16 +291,18 @@ func NewClient(httpClient *http.Client) *Client {
...
 // If a nil httpClient is provided, a new http.Client will be used.
 //
 // Note that NewEnterpriseClient is a convenience helper only;
 // its behavior is equivalent to using NewClient, followed by setting
 // the BaseURL and UploadURL fields.
 //
 // Another important thing is that by default, the GitHub Enterprise URL format
-// should be http(s)://[hostname]/api/v3 or you will always receive the 406 status code.
+// should be http(s)://[hostname]/api/v3/ or you will always receive the 406 status code.
+// The Upload URL format should be http(s)://[hostname]/api/uploads/.
```
Comment (nightlark) [inline github/github.go:305]: Just a nit, matching with the case for other uses of `upload URL`.
```suggestion
// The upload URL format should be http(s)://[hostname]/api/uploads/.
```

**Comment by gmlewis** [comment]
Thank you, @nightlark!
Merging.

**Pull request merged**